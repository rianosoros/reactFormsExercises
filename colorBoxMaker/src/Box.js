import React from 'react';

function box({ id,
  width = 5,
  height = 5,
  color = 'white',
  handleRemove 
}) {
  const removeBox= () => {handleRemove(id);};

  return (
    <div
    style={{
      width: `${width}em`,
      height: `${height}em`,
      backgroundColor: color,
      margin: '10px',
      position: 'relative',
      border: '1px solid black', 
    }}
    >
      <button onClick={removeBox} style={{ position: 'absolute', top: '5px', right: '5px' }}>
        X
      </button>
    </div>
  );
}

export default box;
