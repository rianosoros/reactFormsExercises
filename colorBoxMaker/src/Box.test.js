import { render } from '@testing-library/react';
import Box from './Box';
import React from 'react';

test('renders', () => {
  render(<Box />);
});

test('matches snapshot', () => {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
