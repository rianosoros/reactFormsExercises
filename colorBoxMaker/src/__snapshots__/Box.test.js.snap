// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 5em; height: 5em; background-color: white; margin: 10px; position: relative; border: 1px solid black;"
  >
    <button
      style="position: absolute; top: 5px; right: 5px;"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
